lc0/CMakeLists.txt

```diff
@@ -0,0 +1,71 @@
+# This file is part of Leela Chess Zero.
+# Copyright (C) 2018 The LCZero Authors
+# 
+# Leela Chess is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# Leela Chess is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+
+project (lc0 CXX C)
+cmake_minimum_required (VERSION 3.1)
+
+
+#-------------------------------------------------------------------------------
+# General Settings
+#-------------------------------------------------------------------------------
+set (CMAKE_CXX_STANDARD 14)
+
+if (NOT CMAKE_BUILD_TYPE)
+  set (CMAKE_BUILD_TYPE "Release")
+endif (NOT CMAKE_BUILD_TYPE)
+
+# See if we can set optimization flags as expected.
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
```

Comment:
Would it be possible to additionally enable `-Wthread-annotations` when compiling with clang? Thread annotations are really useful in multithreaded code.